features/minion/heartbeat/consumer/src/main/java/org/opennms/minion/heartbeat/consumer/HeartbeatConsumer.java

```diff
@@ -142,31 +148,43 @@ public void handleMessage(MinionIdentityDTO minionHandle) {
 
         minionDao.saveOrUpdate(minion);
 
-        if (prevLocation == null) {
-            final EventBuilder eventBuilder = new EventBuilder(EventConstants.MONITORING_SYSTEM_ADDED_UEI,
-                    "OpenNMS.Minion.Heartbeat");
-            eventBuilder.addParam(EventConstants.PARAM_MONITORING_SYSTEM_TYPE, OnmsMonitoringSystem.TYPE_MINION);
-            eventBuilder.addParam(EventConstants.PARAM_MONITORING_SYSTEM_ID, minionHandle.getId());
-            eventBuilder.addParam(EventConstants.PARAM_MONITORING_SYSTEM_LOCATION, nextLocation);
-            try {
-                eventProxy.send(eventBuilder.getEvent());
-            } catch (final EventProxyException e) {
-                throw new DataAccessResourceFailureException("Unable to send event", e);
+        // Provision the minions node in a separate thread.
+        final OnmsMinion onmsMinion = minion;
+
+        executor.execute(() -> {
```

Comment:
We don't actually speed up the processing of these in any way, so I would expect that these tasks will continue to pile on to the executor queue.

Can we short circuit the call to provision by verifying if the Minion node already exists? This should avoid the processing delays caused by un-marshalling potentially large requisitions.

Reply:
Yeah, makes sense. Addressed that.